libqpdf/QPDF_encryption.cc

```diff
@@ -1046,7 +1046,7 @@ QPDF::decryptStream(
     Pipeline*& pipeline,
     QPDFObjGen const& og,
     QPDFObjectHandle& stream_dict,
-    std::vector<std::shared_ptr<Pipeline>>& heap)
+    std::unique_ptr<Pipeline>& heap)
```

Comment:
Now it's not a heap anymore. It can only hold one item. Maybe change the name of the variable. I'd call it `decrypt_pipeline` and put a comment in the implementation stating that the caller needs to own the pointer to ensure that the pipeline lives until it's finished being used. (Where are rust lifetimes when you need them?)

Reply:
Will do.

Reply:
Updated.